applications/GeoMechanicsApplication/custom_conditions/T_condition.cpp

```diff
@@ -0,0 +1,173 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//
+//  Main authors:    Mohamed Nabi
+//
+
+
+// Application includes
+#include "custom_conditions/T_condition.hpp"
+
+namespace Kratos
+{
+
+template<unsigned int TDim, unsigned int TNumNodes>
+Condition::Pointer TCondition<TDim,TNumNodes>::Create(
+    IndexType NewId,
+    NodesArrayType const& ThisNodes,
+    PropertiesType::Pointer pProperties) const
+{
+    return Condition::Pointer(new TCondition(NewId, GetGeometry().Create(ThisNodes), pProperties));
+}
+
+// ============================================================================================
+// ============================================================================================
+template<unsigned int TDim, unsigned int TNumNodes>
+void TCondition<TDim,TNumNodes>::GetDofList(
+    DofsVectorType& rConditionDofList,
+    const ProcessInfo& rCurrentProcessInfo) const
+{
+    KRATOS_TRY
+
+    const GeometryType& rGeom = GetGeometry();
+    const unsigned int nDof = 1;
```

Comment:
You can use constexpr to define nDof and consitionSize

Reply:
Then both nDof and conditionsSize can be moved to the header file and do not need to be defined in each function

Reply:
nDof and conditionSize is already moved to the header file. However, I found constexpr gives compiliation error. Then after googling, I reached to conclusion that constexpr works only for static variables, if I understood it correctly.

Reply:
I think you need to write it as `static constexpr unsigned int nDof =1;`